### DISCUSSION TITLE
Proposal to refocus the language on Web, IoT and Security while remaining general-purpose

### DISCUSSION THREAD

**Discussion by mcfriend99** [zuri-lang/zuri#163] [General]
## TL;DR

I'm raising a proposal to refocus Blade language and its continued development on built-in support and focus on Web, IoT and secure application development without needing to install any third-party package and without losing it's ability for general-purpose development.

## Content

After multiple considerations, I begin to feel that while being a general-purpose language, Blade will benefit a lot more from crafting a niche for itself as a general-purpose batteries-included language with a focus on Web, IoT and secure application development where third-party packages are completely optional.

In order to achieve this, Blade will need to begin to provide tooling to support this goals. This will include at a bare minimum the following tools:

- Built-in testing framework.
- Built-in ORM support.
- Built-in support for more databases.
- Built-in support for media processing (Image, audio, video etc).
- Built-in templating engine.
- Built-in routing library.
- Built-in cryptographic library.
- A more advanced/robust HTTP server implementation (we'll need to go beyond basics and aspire to achieve built-in production grade http server in the `http` module with support for UDP).
- A more robust http client implementation in the `http` module.
- Built-in device integrations (such as support for COM/Ports, USB etc).
- Built-in mail library.
- Built-in package management library (it will encourage adoption for Blade).
- Built-in C extension builder/packager.

Just to mention a few.

Along with this proposal, I have proposed some major changes to the Blade distribution.

1. Merge [Nyssa](https://github.com/blade-lang/nyssa) into Blade as the built-in package manager.
2. Merge [qi](https://github.com/mcfriend99/qi) into Blade as the built-in testing framework.
3. Merge [wire](https://github.com/mcfriend99/wire) into Blade as the built-in template engine.
4. Merge [kite](https://github.com/blade-lang/kite) into Blade as the default C extension builder/packager.
5. Merge [miniroute](https://github.com/BenStigsen/miniroute) as the built-in router.
6. Merge [mysql](https://github.com/mcfriend99/mysql) int Blade as the built-in MySQL database integration.

## What does this mean for the next release.

This proposal is aimed to start taking effect at version `0.0.86`. The next release should be unaffected by this to allow more time for bug-fixes and testing for the proposed mergers.

## Who can vote?

Everybody.

## How to Vote

To vote on this proposal, kindly drop a comment or a like.

:heart_on_fire: @mcfriend99

**Comment by benstigsen** [comment]
I support this! I think that Blade is great for web-development, which is personally what I see myself using it for.
Out of all the 6 major changes listed, I think Nyssa being shipped with Blade by default is the most important one.

**Comment by mcfriend99** [comment]
As `v0.0.85` has been released, I've gone ahead to create the project [Odyssey](https://github.com/orgs/blade-lang/projects/1) to manage the implementation of this proposal. These changes will be implemented in the branch [odyssey](https://github.com/blade-lang/blade/tree/odyssey) and all PR for this project should target that branch.
